### ISSUE TITLE
Discussion: Test Plan and ReadMe Documentation

### ISSUE THREAD

**Issue by blueworrybear** [nexfoundation/vanilla#132]
**Describe the Feature**

We have started NEX forum development since **Oct. 2020**.
There are:

1.  **155** commits
2.  **129** files changed

However, there is **zero** files changed relating to testing.
It's dangerous if we keep moving without any automation tests.
We may keep adding new features, which may break some existed codes.
Therefore, I suggest that we have to add more tests, which focus on:

1. PHPUnit tests
2. Jest tests

The problem is that we have no information about where to add unit tests.
We need some information, such as code coverage to help us making decision.

TBH, current README page does not provide anything useful for the developers.
The badges on the REAME page are out of date, and there are too many unnecessary information for Vanilla.
Hence, I suggest to update README page and add information below:

1. Information about NEX Forum
    include organization information (ex. logo) , our vision, code of conduct and etc.
2. GitHub Action Status
3. Code Coverage
4. Current Released Version
5. Server Status

In summary, we have to update README page to provide more useful information and make plans for testing.
Please leave your comment here, any feedback is welcome 😄

**Comment by kun-yu-tsai** [comment]
I strongly agree with this especially the necessity of better documentation for developers to overcome the hurdles while doing development. 

After most of the features are settled, let's start a series of discussion about the implementation priority and the final goal we want to achieve.

Thanks again to raise this. As like Benno said, 有什麼想法都可以說唷～～～～很需要每個人的想法，大膽地說出來吧～！
